ddclient.in

```diff
@@ -5170,6 +5185,113 @@ sub nic_dinahosting_update {
     }
 }
 
+######################################################################
+## nic_gandi_examples
+## by Jimmy Thrasibule <dev@jimmy.lt>
+######################################################################
+sub nic_gandi_examples {
+    return <<"EoEXAMPLE";
+o 'gandi'
+
+The 'gandi' protocol is used by the LiveDNS service offered by gandi.net.
+
+Description of Gandi's LiveDNS API can be found at:
+
+    https://api.gandi.net/docs/livedns/
+
+
+Available configuration variables:
+  * password: The Gandi API key. If you don’t have one yet, you can generate
+              your production API key from the API Key Page (in the Security
+              section).
+  * ttl:      The time-to-live value associated with the updated DNS record.
+  * zone:     The DNS zone to be updated.
+
+Example ${program}.conf file entries:
+  ## Single host update.
+  protocol=gandi,                           \\
+  zone=example.com,                         \\
+  password=my-gandi-api-key,                \\
+  ttl=3600                                  \\
+  host.example.com
+
+  ## Multiple host update.
+  protocol=gandi,                           \\
+  zone=example.com,                         \\
+  password=my-gandi-api-key,                \\
+  ttl=3600                                  \\
+  hosta.example.com,hostb.sub.example.com
+EoEXAMPLE
+}
+
+######################################################################
+## nic_gandi_update
+######################################################################
+sub nic_gandi_update {
+    debug("\nnic_gandi_update -------------------");
+
+    # Update each set configured host.
+    foreach my $h (@_) {
+        my $ip = delete $config{$h}{'wantip'};
+        (my $hostname = $h) =~ s/\.\Q$config{$h}{zone}\E$//;
+
+        info("%s -- Setting IP address to %s.", $h, $ip);
+        verbose("UPDATE:", "updating %s", $h);
+
+        my $headers;
+        $headers  = "Content-Type: application/json\n";
+        $headers .= "Authorization: Apikey $config{$h}{'password'}\n";
+
+        my $data = encode_json({
+            items => [{
+                rrset_type   => is_ipv6($ip) ? "AAAA" : "A",
+                rrset_ttl    => $config{$h}{'ttl'},
+                rrset_values => [$ip],
+            }]
+        });
+
+        my $url = "https://$config{$h}{'server'}/$config{$h}{'script'}/livedns/domains/$config{$h}{'zone'}/records/$hostname";
+        my $reply = geturl({
+            proxy   => opt('proxy'),
+            url     => $url,
+            headers => $headers,
+            method  => 'PUT',
+            data    => $data,
+        });
+        unless ($reply) {
+            failed("%s -- Could not connect to %s.", $h, $config{$h}{'server'});
+            next;
+        }
+        next if !header_ok($h, $reply);
```

Comment:
Unfortunately `header_ok()` only accepts 200 but the API docs say it returns 201. Would you change `header_ok()` to accept any 2xx code?

```perl
        if ($result =~ /^2\d\d$/) {
```